services/QuillLMS/app/workers/assign_recommendations_worker.rb

```diff
@@ -6,17 +6,15 @@ class AssignRecommendationsWorker
 
   # rubocop:disable Metrics/CyclomaticComplexity
   def perform(options={})
-    options = options.with_indifferent_access
-
-    assign_on_join = options[:assign_on_join] || false
-    assigning_all_recommendations = options[:assigning_all_recommendations] || false
-    classroom_id = options[:classroom_id]
-    is_last_recommendation = options[:is_last_recommendation]
-    lesson = options[:lesson]
-    order = options[:order]
-    pack_sequence_id = options[:pack_sequence_id]
-    student_ids = options[:student_ids]
-    unit_template_id = options[:unit_template_id]
+    assign_on_join = options['assign_on_join'] || false
+    assigning_all_recommendations = options['assigning_all_recommendations'] || false
+    classroom_id = options['classroom_id']
+    is_last_recommendation = options['is_last_recommendation']
```

Comment:
This is likely outside of the scope of this PR, but the code in this job after `return unless is_last_recommendation` might better fit in the the batch callback you just created (i.e. it looks like the intention of that code is a jury-rigged version of this new batch code).